pkg/localstorev2/localstore_test.go

```diff
@@ -0,0 +1,508 @@
+// Copyright 2023 The Swarm Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package localstore_test
+
+import (
+	"context"
+	"errors"
+	"io/ioutil"
+	"os"
+	"testing"
+
+	localstore "github.com/ethersphere/bee/pkg/localstorev2"
+	pinstore "github.com/ethersphere/bee/pkg/localstorev2/internal/pinning"
+	"github.com/ethersphere/bee/pkg/localstorev2/internal/upload"
+	chunktesting "github.com/ethersphere/bee/pkg/storage/testing"
+	storage "github.com/ethersphere/bee/pkg/storagev2"
+	"github.com/ethersphere/bee/pkg/swarm"
+)
+
+func verifySessionInfo(
+	t *testing.T,
+	repo *storage.Repository,
+	sessionID uint64,
+	chunks []swarm.Chunk,
+	has bool,
+) {
+	t.Helper()
+
+	for _, ch := range chunks {
+		hasFound, err := repo.ChunkStore().Has(context.TODO(), ch.Address())
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		if hasFound != has {
+			t.Fatalf("unexpected chunk state, exp has chunk %t got %t", has, hasFound)
```

Comment:
The recommended format for test messages is: `"msg: want %v, have %v"`, so in this case: `"unexpected chunk state: want %t, have %t"`. The same applies below.